manifests/init.pp

```diff
@@ -287,7 +287,9 @@
   # logging
   $debug                       = $etcd::params::debug,
   $log_package_levels          = $etcd::params::log_package_levels,
-  $journald_forward_enable     = $etcd::params::journald_forward_enable
+  $journald_forward_enable     = $etcd::params::journald_forward_enable,
+  # metrics
+  $metrics =                   = $etcd::params::metrics
```

Comment:
So, now this `$metrics` variable has `basic` as the default value and can be overridden by whoever instantiates `etcd` class or via the hiera.
But I don't see that it's used anywhere.
Don't you need to set that `ETCD_METRICS` environment variable (as you did in the original diff) to actually change behaviour of `etcd` according to the value of `$metrics` variable?

Reply:
I am overwriting the metric parameter here https://github.yelpcorp.com/sysgit/puppet/pull/4558, Also the metric option is passed as cli parameter, so we dont need the environment variable

Reply:
:+1: